sway_libs/src/nft/extensions/burnable/burnable.sw

```diff
@@ -0,0 +1,50 @@
+library burnable;
+
+dep burnable_events;
+
+use burnable_events::BurnEvent;
+use ::nft::{errors::{AccessError, InputError}, nft_core::NFTCore, nft_storage::{BALANCES, TOKENS}};
+use std::{auth::msg_sender, hash::sha256, logging::log, storage::{get, store}};
+
+pub trait Burnable {
+    /// Deletes this token from storage and decrements the balance of the owner.
+    ///
+    /// * Reverts
+    ///
+    /// * When sender is not the owner of the token.
+    #[storage(read, write)]
+    fn burn(self);
+}
+
+impl Burnable for NFTCore {
+    #[storage(read, write)]
+    fn burn(self) {
+        require(self.owner == msg_sender().unwrap(), AccessError::SenderNotOwner);
+
+        let owner = self.owner;
+        let token_id = self.token_id;
```

Comment:
IMO it's a little confusing to assign variables here and then not use them immediately below where they could be used.  It also seems like they are only defined in order to make use of the field short hand syntax (or whatever it is called) ... not sure how I feel about that